BUILD.md

```diff
@@ -0,0 +1,228 @@
+at Getting Started - Installing Git: <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>.
+
+Open up a terminal and run the following:
+
+```bash
+   $ cd $HOME
+   $ mkdir sawtooth
+   $ cd sawtooth
+   $ git clone https://github.com/hyperledger/sawtooth-core.git
+```
+
+Step Three: Initializing Your Environment
+-------------
+
+The Sawtooth build and test infrastructure requires dependencies be built.
+These dependencies include the protocol buffers definitions (under the protos
+dir) be built for the target languages and a set of docker images be built with
+with the required build and runtime dependencies installed.  To build all the
+dependencies for all of the components of Sawtooth, run this command:
+
+```bash
+  $ bin/build_all
+```
+
+By default, this builds all the components for all the components and sdks in
+the repository. This is a very slow process. It is recommended that you only
+build the language dependencies you need for your development purposes. The
+minimum requirements for running the Sawtooth validator is the python build:
+
+```bash
+  $ bin/build_all -l python
+```
+
+To get details on all the options for the build_all script
+
+```bash
+  $ bin/build_all -h
+```
+
+If you are working on the core validator, only the ‘python’ language is
+required. If you are working on a particular language sdk then you will need to
+build the language that sdk is for as well.
+
+**Note:** This build environment uses docker to virtualize the build and
+execute the code in the development directory. This allows for modification
+made to the local source to be built and tested without installing local
+dependencies on your machine.
+
+**Note:**  If you wish to configure your development machine to do compilation
+directly on the host without a docker virtualization, the docker files in the
+sawtooth-core/docker. For example, the file sawtooth-core/docker
+sawtooth-dev-python describes the configuration and components needed to build
+and run the python components on a system.
+
+Step Four: Development Tasks
+-------------
+
+To run a full Validator Node from the local source:
+
+```bash
+  $ docker-compose -f docker/compose/sawtooth-local.yaml up
+```
+
+This will start a validator with the following components attached to it:
+  - Xo Transaction Processor ( python implementation )
+  - IntKey Transaction Processor ( python implementation )
+  - Settings Transaction Processor
+  - REST API - available on host port 8008
+  - Shell - for running sawtooth cli commands
+
+From another console window you can access the shell
+```bash
+  $ docker-compose -f docker/compose/sawtooth-local.yaml exec client bash
+```
+
+This command uses Docker Compose and the development docker images. These
+docker images have the runtime dependencies installed in them but run sawtooth
+from the source in your workspace. Inspecting
+`docker/compose/sawtooth-local.yaml` will show how the various component
```

Comment:
will show how the various component??
Don't leave me hanging. I'm really curious what it will show me. ;)